packages/block-library/src/gallery/index.php

```diff
@@ -32,12 +32,53 @@ function block_core_gallery_data_id_backcompatibility( $parsed_block ) {
 
 add_filter( 'render_block_data', 'block_core_gallery_data_id_backcompatibility' );
 
+/**
+ * Adds a style tag for the --wp--style--unstable-gallery-gap var.
+ *
+ * The Gallery block needs to recalculate Image block width based on
+ * the current gap setting in order to maintain the number of flex columns
+ * so a css var is added to allow this.
+ *
+ * @param array  $attributes Attributes of the block being rendered.
+ * @param string $content Content of the block being rendered.
+ * @return string The content of the block being rendered.
+ */
+function block_core_gallery_render( $attributes, $content ) {
+	$gap = _wp_array_get( $attributes, array( 'style', 'spacing', 'blockGap' ) );
+	// Skip if gap value contains unsupported characters.
+	// Regex for CSS value borrowed from `safecss_filter_attr`, and used here
+	// because we only want to match against the value, not the CSS attribute.
+	$gap       = preg_match( '%[\\\(&=}]|/\*%', $gap ) ? null : $gap;
+	$id        = uniqid();
+	$class     = 'wp-block-gallery-' . $id;
```

Comment:
I just caught this. This should rather have been:

```php
$class     = wp_unique_id( 'wp-block-gallery-' );
```

See https://github.com/WordPress/gutenberg/issues/38889 and #38891

Reply:
I've applied this change in #39983.

Reply:
Thanks for this fix @westonruter